certora/harnesses/GuardianStorageHarness.sol

```diff
@@ -5,11 +5,12 @@ import {GuardianStorage} from "../../contracts/modules/social_recovery/storage/G
 contract GuardianStorageHarness is GuardianStorage {
 
     /**
-     * @dev Gets the count of guaridans for a wallet.
+     * @notice Gets the count of guaridans for a wallet based on the actual linked list.
+     * @dev This is needed for FV to avoid cases where we have values which does not start with SENTINEL.
```

Comment:
This doesn't make sense to me... doesn't the linked list *always* start with `SENTINAL` by definition?

To me this is really about integrity of `guardiansCount`.

Reply:
> doesn't the linked list always start with SENTINAL by definition?

Yes, but with FV we can have initial global state which can have guardians with connections to other addresses, which does not start with `SENTINAL`. So, with this function, we are only taking into account the guardians which can be counted from the `SENTINAL`.

Reply:
Oh, you mean hidden guardians that are not part of the linked list?

Reply:
Yes

Reply:
You can take some inspiration from `safe-smart-account's` linked list verification: https://github.com/safe-global/safe-smart-account/blob/main/certora/specs/ModuleReach.spec

It's (crazy) complex, so it might be worth it to ask Jochen for help.

Reply:
So, this is only until we have the invariant rules up and running for the GuardianStorage. Once that is running, we will just `requireInvariant` the relevant ones.

Reply:
I still stand by my point:

> But, I don't think this rule helps at all though. You can have other refernce loops that start with `address(2)` and I don't understand how this function will help filter those out...

Reply:
Slightly modified the comment.

Reply:
My issue is with this statement:

> This is needed for FV to avoid cases where we have values which does not start with SENTINEL.

I don't understand how this function helps with this at all.

Reply:
What it intends to say is, that it only counted guardians which can be linked from the `SENTINEL`, and avoided the rest of the values.

Reply:
I would say then that this function:

> Counts the guardians by iterating through the linked list starting at the sentinel address, instead of relying on the `count` storage variable.

Reply:
Updated the NatSpec 👍🏾